src/transformers/generation/utils.py

```diff
@@ -2516,19 +2550,19 @@ def generate(
                     f"assisted generation is not supported with stateful models, such as {self.__class__.__name__}"
                 )
 
-            # 11. Get the candidate generator, given the parameterization
+            # 10. Get the candidate generator, given the parameterization
             candidate_generator = self._get_candidate_generator(
                 generation_config=generation_config,
                 input_ids=input_ids,
                 inputs_tensor=inputs_tensor,
                 assistant_model=assistant_model,
                 logits_processor=logits_processor,
-                target_tokenizer=tokenizer,
-                assistant_tokenizer=assistant_tokenizer,
+                target_tokenizer=gen_mode_kwargs.get("tokenizer"),
+                assistant_tokenizer=gen_mode_kwargs.get("assistant_tokenizer"),
                 model_kwargs=model_kwargs,
```

Comment:
```suggestion
               model_kwargs=model_kwargs,
               **gen_mode_kwargs
```
If `gen_mode_kwargs` contains keys that are not used, it will an exception -> this is wanted